packages/integration-karma/test/shadow-dom/EventTarget-methods/EventTarget.addEventListener.spec.js

```diff
@@ -159,4 +159,77 @@ describe('EventTarget.addEventListener', () => {
             expect(id).toEqual('second-container');
         });
     });
+    // TODO [#2253]: Enable in all modes once bug is fixed
+    if (process.env.NATIVE_SHADOW) {
+        describe('identical listeners', () => {
+            describe('should discard multiple identical listeners on same target', () => {
+                function testDiscardingIdenticalListeners(
+                    target,
+                    listenerTargets,
+                    expected,
+                    composed = false
+                ) {
+                    const logs = [];
+                    const listener = function (event) {
+                        logs.push([this, event.currentTarget]);
+                    };
+                    listenerTargets.forEach((target) => {
+                        target.addEventListener('dedupe', listener);
+                        target.addEventListener('dedupe', listener); // identical listener
+                    });
+                    target.dispatchEvent(new CustomEvent('dedupe', { bubbles: true, composed }));
+                    expect(logs).toEqual(expected);
+                }
+                it('listeners on shadow root, non-composed event', () => {
+                    testDiscardingIdenticalListeners(
+                        nodes.button,
+                        [nodes.button, nodes['container_div'], nodes['x-container'].shadowRoot],
+                        [
+                            [nodes.button, nodes.button],
+                            [nodes['container_div'], nodes['container_div']],
+                            [nodes['x-container'].shadowRoot, nodes['x-container'].shadowRoot],
+                        ]
+                    );
+                });
+
+                it('listeners on host, composed event', () => {
+                    testDiscardingIdenticalListeners(
+                        nodes.button,
+                        [nodes.button, nodes['container_div'], nodes['x-container']],
+                        [
+                            [nodes.button, nodes.button],
+                            [nodes['container_div'], nodes['container_div']],
+                            [nodes['x-container'], nodes['x-container']],
+                        ],
+                        true
+                    );
+                });
+            });
+            describe('should invoke identical listeners on same target with different options', () => {
+                // Note: addEventListener() with options are restricted for shadow root and host elements
```

Comment:
For bonus points, write the tests for shadow roots and hosts, and add an expectation of an exception. Then, when we open that up in the future, we'll have the coverage since we'd have to fix the test.